File: cmd/routing/producer.go
package routing

import (
	"rabbitmq/utils"

	"github.com/spf13/cobra"
	"github.com/streadway/amqp"
)

var Producer = &cobra.Command{
	Use:   "routing_producer",
	Short: "Routing producer",
	Run: func(cmd *cobra.Command, args []string) {
		conn, err := amqp.Dial(utils.Env.AMQP_URL)
		utils.LogError(err, "Can not connect to amqp server")
		defer conn.Close()

		ch, err := conn.Channel()
		utils.LogError(err, "Can not create a channel")
		defer ch.Close()

		err = ch.ExchangeDeclare("routing", amqp.ExchangeDirect, false, false, false, false, nil)
		utils.LogError(err, "Can not declare a exchange")

		err = ch.Publish("routing", "analytics_only", false, false, amqp.Publishing{Body: []byte("Data should be analysed.")})
		utils.LogError(err, "Can not publish message to channel")

		err = ch.Publish("routing", "payment_only", false, false, amqp.Publishing{Body: []byte("Buy some butter.")})
		utils.LogError(err, "Can not publish message to channel")
	},
}

File: src/routing/producer.ts
import { connect } from "amqplib";

const producer = async () => {
    const connection = await connect({
        hostname: process.env.AMQP_HOSTNAME,
        port: parseInt(process.env.AMQP_PORT as string) || 5672,
        username: process.env.AMQP_USERNAME,
        password: process.env.AMQP_PASSWORD
    });

    const channel = await connection.createChannel();
    await channel.assertExchange("routing", "direct", { autoDelete: false, durable: false })
    channel.publish("routing", "analytics_only", Buffer.from("Data should be analysed."))
    channel.publish("routing", "payment_only", Buffer.from('Buy some butter'));
    await channel.close()
    await connection.close()
}

export default producer;